packages/slice-machine/src/tracker.ts

```diff
@@ -0,0 +1,156 @@
+      if (!!this.#client) return;
+      this.#client = await AnalyticsBrowser.standalone(segmentKey);
+    } catch (error) {
+      // If the client is not correctly setup we are silently failing as the tracker is not a critical feature
+      console.warn(error);
+    }
+  }
+
+  /** Private methods **/
+
+  #trackEvent(
+    eventType: AllSliceMachineEventType,
+    attributes: Record<string, unknown> = {}
+  ): void {
+    if (!this.#isTrackingPossible(this.#client)) {
+      return;
+    }
+
+    this.#client
+      .track(eventType, attributes)
+      .catch(() =>
+        console.warn(`Couldn't report event ${eventType}: Tracking error`)
+      );
+  }
+
+  #identify(userId: string): void {
+    if (!this.#isTrackingPossible(this.#client)) {
+      return;
+    }
+
+    this.#client
+      .identify(userId)
+      .catch(() => console.warn(`Couldn't report identify: Tracking error`));
+  }
+
+  #group(attributes: Record<string, unknown> = {}): void {
+    if (!this.#isTrackingPossible(this.#client)) {
+      return;
+    }
+
+    if (!this.#repoName) {
+      return;
+    }
+
+    try {
+      this.#client.group(this.#repoName, attributes);
+    } catch {
+      console.warn(`Couldn't report group: Tracking error`);
+    }
+  }
+
+  #isTrackingPossible(
+    client: ClientAnalytics | null
+  ): client is ClientAnalytics {
+    return this.#isTrackingActive && !!client;
+  }
+
+  /** Public methods **/
+
+  identifyUser(userId: string): void {
+    this.#identify(userId);
+  }
+
+  groupLibraries(libs: readonly LibraryUI[], version: string): void {
+    const downloadedLibs = libs.filter((l) => l.meta.isDownloaded);
+
+    this.#group({
+      manualLibsCount: libs.filter((l) => l.meta.isManual).length,
+      downloadedLibsCount: downloadedLibs.length,
+      npmLibsCount: libs.filter((l) => l.meta.isNodeModule).length,
+      downloadedLibs: downloadedLibs.map((l) => l.meta.name || "Unknown"),
+      slicemachineVersion: version,
+    });
+  }
+
+  trackReview(framework: Frameworks, rating: number, comment: string): void {
+    this.#trackEvent(EventType.Review, { rating, comment, framework });
+  }
+
+  trackSlicePreviewSetup(framework: Frameworks, version: string): void {
+    this.#trackEvent(EventType.SlicePreviewSetup, { version, framework });
+  }
+
+  trackOpenSlicePreview(framework: Frameworks, version: string): void {
+    this.#trackEvent(EventType.SlicePreviewOpen, { version, framework });
+  }
+
+  trackOnboardingStart(): void {
+    this.#trackEvent(EventType.OnboardingStart);
+  }
+
+  trackOnboardingContinue(
+    continueOnboardingEventType: ContinueOnboardingType
+  ): void {
+    this.#trackEvent(continueOnboardingEventType);
+  }
+
+  trackOnboardingSkip(screenSkipped: number): void {
+    this.#trackEvent(EventType.OnboardingSkip, {
+      screenSkipped,
+    });
+  }
+}
+
+const Tracker = (() => {
+  let smTrackerInstance: SMTracker;
+
+  const init = () => new SMTracker();
+
+  return {
+    get() {
```

Comment:
By seeing this implem, it looks good but maybe the init part we could simplify it maybe this way: 
```
return {
  get() {
    return smTrackerInstance
  },
  init(param1, param2, param3) {
    if (!instance) new Tracker(param1, param2, param3)
  },
}
```

so we could remove the initialize function, move everything in the constructor and directly call Tracker.init()
The benefit would be a simpler syntax but also setup the class with the right params right away instead of default null values to wait for the initialize to be executed. WDYT?

Reply:
Done

Reply:
The benefit of the singleton is to abstract the instance creation, so if get return null or SMTracker we loose this benefit. That also mean that in the component we should listen for tracker changes to get the moment when the Tracker will be initialize. 

````
useEffect(() => {
    if (tracker) {
      tracker.Track.SlicePreview({ framework, version });    }
  }, [tracker]);
`